### DISCUSSION TITLE
C# Language server in Pulsar (Omnisharp) not working?

### DISCUSSION THREAD

**Discussion by L0grus** [pulsar-edit/.github#197] [Community Packages]
Doesn't Omnisharp work with Pulsar? As i see, it is not updated for very long (2017-2018 year),and launching it in Pulsar  encountering bunch of errors, but, maybe there is a way  to work with it? Or ? it needs complete rewrite to Pulsar?

**Comment by Daeraxa** [comment]
Would this be the [omnisharp-atom](https://web.pulsar-edit.dev/packages/omnisharp-atom) package? Are the errors you are seeing the same as shown on the repo's [issues](https://github.com/OmniSharp/omnisharp-atom/issues/1037)? In which case it seems the issue isn't with Pulsar but was broken before Atom was even sunset.

Otherwise there is work going on to create and support a "generic-lsp" client which will hopefully eliminate the need to require a package for each version to get LSP functionality.

**Reply by Nine-H** [reply]
is the work a continuation of `atom-languageclient` or is there a new package, repo, or branch that people looking for lsp should follow?

**Reply by Daeraxa** [reply]
The generic-lsp package I mentioned is its own thing in development by one of the Pulsar team members, I'm not actually sure as to the easiest way to install it at this point (PPR doesn't like monorepos yet). @mauricioszabo might be able to help out as its author.

Some of the atom-community projects like `atom-languageclient` have a good chance of being moved into the upcoming pulsar-cooperative repos if nobody picks them up or publishes them.
